project-addons/custom_account/account.py

```diff
@@ -359,6 +360,16 @@ def write(self, vals):
                         write({'blocked': inv.payment_term.blocked})
         return res
 
+    @api.multi
+    def invoice_validate(self):
+        res = super(AccountInvoice, self).invoice_validate()
+        for inv in self:
+            invoices_line = self.env['account.invoice.line'].search(
+                [('invoice_id', '=', inv.id)])
+            for inv_line in invoices_line:
```

Comment:
for inv_line in inv.invoice_line: 
El search no hace falta ya que el propio objeto de factura tiene un campo de nombre "invoice_line" que es una one2many contra sus lineas